src/test/java/com/networknt/schema/regex/Issue814Test.java

```diff
@@ -8,40 +8,50 @@ class Issue814Test {
 
     @Test
     void jdkTypePattern() {
-        JDKRegularExpression ex = new JDKRegularExpression("^list|date|time|string|enum|int|double|long|boolean|number$");
+        JDKRegularExpression ex = new JDKRegularExpression("^(list|date|time|string|enum|int|double|long|boolean|number)$");
```

Comment:
As mentioned by @kool79 in #872, if a user wishes only to match these words, then the parentheses are required.